ccl/importccl: TestImportComputed failed

ccl/importccl.TestImportComputed [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2900755&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2900755&tab=artifacts#/) on master @ [f451dd080c95fa73f7d796e1dc8bd6dd33eba09b](https://github.com/cockroachdb/cockroach/commits/f451dd080c95fa73f7d796e1dc8bd6dd33eba09b):


```
Previous write at 0x00c0074df910 by goroutine 1025:
  github.com/cockroachdb/cockroach/pkg/sql/catalog/schemaexpr.(*nameResolver).addColumn()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/schemaexpr/expr.go:265 +0x1d4
  github.com/cockroachdb/cockroach/pkg/sql/catalog/schemaexpr.MakeComputedExprs()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/schemaexpr/computed_column.go:285 +0x8e6
  github.com/cockroachdb/cockroach/pkg/sql/row.NewDatumRowConverter()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/row/row_converter.go:435 +0x1550
  github.com/cockroachdb/cockroach/pkg/ccl/importccl.makeDatumConverter()
      /go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/read_import_base.go:449 +0x18c
  github.com/cockroachdb/cockroach/pkg/ccl/importccl.(*parallelImporter).importWorker()
      /go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/read_import_base.go:653 +0x7c
  github.com/cockroachdb/cockroach/pkg/ccl/importccl.runParallelImport.func1.1()
      /go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/read_import_base.go:553 +0xae
  github.com/cockroachdb/cockroach/pkg/util/ctxgroup.GroupWorkers.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:175 +0x5d
  github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:166 +0x4e
  golang.org/x/sync/errgroup.(*Group).Go.func1()
      /go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57 +0x94

Goroutine 558 (running) created at:
  golang.org/x/sync/errgroup.(*Group).Go()
      /go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:54 +0x73
  github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:165 +0xe4
  github.com/cockroachdb/cockroach/pkg/util/ctxgroup.GroupWorkers()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:175 +0xaa
  github.com/cockroachdb/cockroach/pkg/ccl/importccl.runParallelImport.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/read_import_base.go:552 +0x2f1
  github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:166 +0x4e
  golang.org/x/sync/errgroup.(*Group).Go.func1()
      /go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57 +0x94

Goroutine 1025 (finished) created at:
  golang.org/x/sync/errgroup.(*Group).Go()
      /go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:54 +0x73
  github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:165 +0xe4
  github.com/cockroachdb/cockroach/pkg/util/ctxgroup.GroupWorkers()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:175 +0xaa
  github.com/cockroachdb/cockroach/pkg/ccl/importccl.runParallelImport.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/read_import_base.go:552 +0x2f1
  github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:166 +0x4e
  golang.org/x/sync/errgroup.(*Group).Go.func1()
      /go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57 +0x94
==================
    testing.go:1038: race detected during execution of test
    --- FAIL: TestImportComputed/import-table-avro (1.39s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestImportComputed PKG=./pkg/ccl/importccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-race -parallel=4
</p>
</p>
</details>
/cc @Anzoteh96
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestImportComputed.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/catalog/schemaexpr/expr.go
+++ b/pkg/sql/catalog/schemaexpr/expr.go
@@ -239,7 +239,7 @@ func newNameResolver(
 		*tn,
 		colinfo.ResultColumnsFromColumns(tableID, cols),
 	)
-	nrc := &nameResolverIVarContainer{cols}
+	nrc := &nameResolverIVarContainer{append(make([]catalog.Column, 0, len(cols)), cols...)}
 	ivarHelper := tree.MakeIndexedVarHelper(nrc, len(cols))
 
 	return &nameResolver{

